### PR TITLE
Missing functions addition

### PR DESCRIPTION
**Pull request by Luhrel**
I added some functions that were not implemented in CSFML.
I also modified sfTexture_bind, with an sfTextureCoordinateType arg.

Modified:
sfTexture_bind (Added sfTextureCoordinateType arg)

Added:
sfTextureCoordinateType
sfContext_isExtensionAvailable
sfContext_getFunction
sfRenderWindow_drawVertexBufferRange
sfRenderTexture_drawVertexBufferRange
sfTcpListener_close

### CODE REVIEW

**Comment by LaurentGomila** [comment]
There's a reason why some of these functions are not implemented. They are either for internal use (when it's the case, they are clearly documented as such) or not needed in C because of the API differences.

Of course, some functions may also simply be missing, most likely those that were added recently 😉

**Comment by Luhrel** [comment]
Yup I know and I didn't implemented those, did I ?
Which funcs (in this PR) aren't needed in C ?

**Comment by LaurentGomila** [comment]
`sfTcpListener_close` is not needed since you just free (`sfTcpListener_destroy`) the instance when you're done with it.

`sfTextureCoordinateType` and where it's used, were initially meant for internal use only. I'm not sure if that changed since then.

**Comment by Luhrel** [comment]
> `sfTcpListener_close` is not needed since you just free (`sfTcpListener_destroy`) the instance when you're done with it.

Ok, but should we add it anyway for the other SFML bindings ?

> `sfTextureCoordinateType` and where it's used, were initially meant for internal use only. I'm not sure if that changed since then.

The doc just says : 

> [...] [sf::CoordinateType::Pixels] is used internally by the graphics classes of SFML, it makes the definition of texture coordinates more intuitive for the high-level API, users don't need to compute normalized values.

So it may be used.

**Comment by LaurentGomila** [comment]
> Ok, but should we add it anyway for the other SFML bindings (which are based on CSFML) ?

Nop.

**Comment by silverweed** [comment]
@LaurentGomila how about the VertexBuffer functions? Would they be accepted if they were isolated in their own PR?

**Comment by LaurentGomila** [comment]
Is `sf::VertexBuffer` part of the latest version, or does it have to be released yet (I really don't remember 😁)?

**Comment by silverweed** [comment]
It's surely there in SFML 2.5 :D

**Comment by LaurentGomila** [comment]
Then OK, it should be in CSFML and SFML.Net too.

**Comment by silverweed** [comment]
Ok, cool :)
@Luhrel, would you rather split your PR or let me reopen mine (#153)?

**Comment by Luhrel** [comment]
@silverweed There's no need to split/reopen. All the functions in this PR is for the 2.5.

@LaurentGomila Which name do you prefer `sfRenderTexture_drawVertexBufferWithIndex` or `sfRenderTexture_drawVertexBufferRange` ?

**Comment by LaurentGomila** [comment]
I'd say `sfRenderTexture_drawVertexBufferRange`.

**Comment by Luhrel** [comment]
@LaurentGomila Done. Maybe you can merge this now :wink:

**Comment by silverweed** [comment]
@LaurentGomila any news on this?

File: include/SFML/Graphics/RenderTexture.h [hunk 1]
```diff
@@ -223,6 +223,20 @@ CSFML_GRAPHICS_API void sfRenderTexture_drawRectangleShape(sfRenderTexture* rend
...
+/// \param renderTexture render texture object
+/// \param vertexBuffer  Vertex buffer object to draw
+/// \param firstVertex   Index of the first vertex to render
+/// \param vertexCount   Number of vertices to render
+/// \param states        Render states to use for drawing
+///
+////////////////////////////////////////////////////////////
+CSFML_GRAPHICS_API void sfRenderTexture_drawVertexBufferWithIndex(sfRenderTexture* renderTexture,
+                                                  const sfVertexBuffer* object, size_t firstVertex,
+                                                  size_t vertexCount, const sfRenderStates* states);
```
Comment (eXpl0it3r) [inline include/SFML/Graphics/RenderTexture.h:238]: Align with opening parenthesis like the `drawPrimitives` below

File: include/SFML/Graphics/RenderWindow.h [hunk 2]
```diff
@@ -466,6 +466,20 @@ CSFML_GRAPHICS_API void sfRenderWindow_drawRectangleShape(sfRenderWindow* render
...
+/// \param renderWindow render window object
+/// \param vertexBuffer Vertex buffer object to draw
+/// \param firstVertex  Index of the first vertex to render
+/// \param vertexCount  Number of vertices to render
+/// \param states       Render states to use for drawing
+///
+////////////////////////////////////////////////////////////
+CSFML_GRAPHICS_API void sfRenderWindow_drawVertexBufferWithIndex(sfRenderWindow* renderWindow,
+                                                  const sfVertexBuffer* object, size_t firstVertex,
+                                                  size_t vertexCount, const sfRenderStates* states);
```
Comment (eXpl0it3r) [inline include/SFML/Graphics/RenderWindow.h:481]: Align with opening parenthesis like the `drawPrimitives` below

File: src/SFML/Graphics/RenderTexture.cpp [hunk 3]
```diff
@@ -244,6 +244,16 @@ void sfRenderTexture_drawVertexBuffer(sfRenderTexture* renderTexture, const sfVe
 }
 
 
+////////////////////////////////////////////////////////////
+void sfRenderTexture_drawVertexBufferWithIndex(sfRenderTexture* renderTexture,
+                                              const sfVertexBuffer* object, size_t firstVertex,
+                                              size_t vertexCount, const sfRenderStates* states)
```
Comment (eXpl0it3r) [inline src/SFML/Graphics/RenderTexture.cpp:250]: Align with opening parenthesis like the `drawPrimitives` below

**Review by eXpl0it3r — CHANGES_REQUESTED** [review]
Really sorry for the long wait on this 😓 

Can you squash the two vertex buffer commits and fix the small style changes?

**Review by eXpl0it3r — APPROVED** [review]

**Comment by eXpl0it3r** [comment]
Allowed myself to integrate the last commit in the relevant commits and rebase, as the pipeline was slightly outdated 😅

**Comment by eXpl0it3r** [comment]
Thanks a lot for these fixes and the patience. 🥳 

Sorry again it took so long 😞

**Pull request merged**